src/sheet/sheet.c

```diff
@@ -317,6 +316,10 @@ gboolean sheet_get_adjustments (const Sheet *sheet, GtkAdjustment **hadj, GtkAdj
  */
 void sheet_change_zoom (Sheet *sheet, gdouble factor)
 {
+	double zoom;
+	sheet_get_zoom (sheet, &zoom);
+	sheet_set_zoom (sheet, zoom * factor);
+
```

Comment:
No, it doesn't work like that.

Do the following test:

- open a schematic of your choice;
- zoom in three times;
- resize the sheet once;
- the zoom is restored instead of being kept: we don't want this.

Apply the zoom patch (91d6a409652b737894f7b49c108965755ab519b8) and verify the result again: the zoom is kept after resizing.

Please confirm.